google/cloud/storage/internal/curl_handle.cc

```diff
@@ -226,26 +226,36 @@ Status CurlHandle::AsStatus(CURLcode e, char const* where) {
     case CURLE_NOT_BUILT_IN:
       code = StatusCode::kUnknown;
       break;
+
     case CURLE_COULDNT_RESOLVE_PROXY:
     case CURLE_COULDNT_RESOLVE_HOST:
     case CURLE_COULDNT_CONNECT:
-    case CURLE_WEIRD_SERVER_REPLY:
       code = StatusCode::kUnavailable;
       break;
+
+#if LIBCURL_VERSION_NUM >= 0x075100
+    case CURLE_WEIRD_SERVER_REPLY:
+      code = StatusCode::kUnknown;
+      break;
+#endif  // CURL >= 7.51.0
+
     case CURLE_REMOTE_ACCESS_DENIED:
       code = StatusCode::kPermissionDenied;
       break;
+
     case CURLE_FTP_ACCEPT_FAILED:
     case CURLE_FTP_WEIRD_PASS_REPLY:
     case CURLE_FTP_WEIRD_227_FORMAT:
     case CURLE_FTP_CANT_GET_HOST:
-    case CURLE_HTTP2:
+    // case CURLE_HTTP2:  unavailable in old versions of libcurl
```

Comment:
Q: Should be conditionalize on LIBCURL_VERSION_NUM, like above, then?  Or, similarly handled by the default case?

Reply:
See above.